File: devices-api-java-mvc/src/main/java/com/lucas/devicesapijavamvc/api/DeviceMapper.java
package com.lucas.devicesapijavamvc.api;

import com.lucas.devicesapijavamvc.api.dto.DeviceRequest;
import com.lucas.devicesapijavamvc.api.dto.DeviceResponse;
import com.lucas.devicesapijavamvc.domain.Device;

public final class DeviceMapper {

    private DeviceMapper() {}

    public static Device toNewEntity(DeviceRequest req) {
        return Device.createNew(req.name(), req.brand(), req.state());
    }

    public static DeviceResponse toResponse(Device d) {
        return new DeviceResponse(
                d.getId(),
                d.getExternalId(),
                d.getName(),
                d.getBrand(),
                d.getState().name(),
                d.getCreationTime()
        );
    }
}

File: devices-api-kotlin-reactive/src/main/kotlin/com/lucas/devicesapikotlinreactive/infrastructure/api/DeviceMapper.kt
package com.lucas.devicesapikotlinreactive.infrastructure.api

import com.lucas.devicesapikotlinreactive.domain.model.Device
import com.lucas.devicesapikotlinreactive.infrastructure.api.dto.DeviceRequest
import com.lucas.devicesapikotlinreactive.infrastructure.api.dto.DeviceResponse

object DeviceMapper {

    fun toDomain(req: DeviceRequest): Device =
        Device.create(
            name = req.name,
            brand = req.brand,
            state = req.state
        )

    fun toResponse(device: Device): DeviceResponse =
        DeviceResponse(
            id = device.id ?: 0L,
            externalId = device.externalId,
            name = device.name,
            brand = device.brand,
            state = device.state.name,
            creationTime = device.creationTime
        )
}
